docs/design.md

```diff
@@ -91,97 +107,204 @@ This pattern is inspired by [sickle_ui](https://github.com/UmbraLuminosa/sickle_
 By encapsulating a widget inside a function, you save on a lot of boilerplate code and can easily change the appearance of all widgets of a certain type.
 By returning `EntityCommands`, you can easily chain multiple widgets together and insert children into a parent widget.
 
-## Assets
+## Asset Preloading
 
 ### Pattern
 
-Define your assets in an enum so each variant maps to a `Handle`:
+Define your assets in a `Handles` resource that maps their path to a `Handle`.
+If you are definiting the assets in code, add them as constants to the `Handles` struct.
+Otherwise, load them dynamically from e.g. a file.
 
 ```rust
-#[derive(Copy, Clone, Eq, PartialEq, Hash, Reflect)]
-pub enum SpriteKey {
-    Player,
-    Enemy,
-    Powerup,
+#[derive(Resource, Debug, Deref, DerefMut, Reflect)]
+#[reflect(Resource)]
+pub struct ImageHandles(HashMap<String, Handle<Image>>);
+
+impl ImageHandles {
+    pub const KEY_PLAYER: &'static str = "assets/images/player.png";
+    pub const KEY_ENEMY: &'static str = "assets/images/enemy.png";
+    pub const KEY_POWERUP: &'static str = "assets/images/powerup.png";
```

Comment:
Isn't this supposed to omit the leading `assets/`?

Reply:
Yes, thanks!

Reply:
[`c286d6f` (#241)](https://github.com/TheBevyFlock/bevy_quickstart/pull/241/commits/c286d6fcbe094cf3e7dd429e4d10e9f7c255f040)